build.sbt

```diff
@@ -1207,10 +1208,11 @@ lazy val `sbt-plugin` = (project in file("dev") / "sbt-plugin")
       val () = (publishLocal in LocalProject("sbt-scripted-tools")).value
     },
     publishTo := {
+      val old = publishTo.value
       if (isSnapshot.value) {
         // Bintray doesn't support publishing snapshots, publish to Sonatype snapshots instead
         Some(Opts.resolver.sonatypeSnapshots)
-      } else publishTo.value
+      } else old
```

Comment:
This evaluates `publishTo.value` always. Is that the intended behavior? If yes, why?

Reply:
Yes, that's right.  This change is to avoid sbt's "linter" that fails the build load if you call `.value` in a branch (here an `else` branch).  The intent of the linter is to avoid problems of _thinking_ `foo.value` is only evaluated in the branch, when in reality all `.value` invocations are evaluated no matter what, before any code in the setting initialisation code.

Here little is going to change because it's not expected that the `publishTo` task has any negatively impacting side-effects (compared to, say, `publish`).  I think there is another way to avoid the lint warning, but I wasn't sure, and I know this works.